ONSdigital/dp-search-reindex-tracker
# Initialise kafka consumers

# What has changed




The following changes have been made to the Search Data Tracker:

- Updated Search Reindex Tracker to initialise kafka consumers for topics listed below
- Updated Graceful shutdown to include closing of kafka consumers
- Updated Search Reindex Tracker health check to include, kafka topic checkers

Kafka Topics:
1. reindex-requested
2. search-data-imported
3. reindex-task-counts

Consumer Group name is the name of the application, `dp-search-reindex-tracker`

NB. The manifest for the Search Data Tracker, in dp-configs, has already been created as required.

# How to review

Check that the code looks correct.

If you would like to test it locally then do as follows:

- Install and run Kafka locally.
- Create the three Kafka topic locally.
- Create the following event (message) in each of the kafka topics:
He2159149-9f7d-4dff-9bbc-0cfbe9ed708f&ons1658415486674332 snOUfuJMIKabYYvn
- Run the Search Reindex Tracker and see the event being logged three times (once for each of the topics) and handled by the "Hello WOrld" handler three times.

# Who can review

Anyone but me
